File: cpp/3370_smallest_number/smallest_number.cpp
#include <cassert>
#include <tuple>
#include <vector>

class Solution {
 public:
  int smallestNumber(int n) {
    int x = 1;
    while (x < n) {
      x = x * 2 + 1;
    }
    return x;
  }
};

int main() {
  std::vector<std::tuple<int, int>> tests{
      {5, 7},
      {10, 15},
      {3, 3},
  };

  for (auto& [n, ans] : tests) {
    assert(Solution().smallestNumber(n) == ans);
  }
}

File: golang/3370_smallest_number/smallest_number.go
// Package main ...
package main

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func smallestNumber(n int) int {
	x := 1
	for x < n {
		x = x*2 + 1
	}
	return x
}

func main() {
	tests := []struct {
		n   int
		ans int
	}{
		{5, 7},
		{10, 15},
		{3, 3},
	}

	for index, test := range tests {
		assert.Equal(&testing.T{}, test.ans, smallestNumber(test.n), index)
	}
}
